server/scripts/populate.py

```diff
@@ -0,0 +1,150 @@
+                return None
+        return club_generator(clubList)
+
+
+def verify(entity, fields, typeof):
+    for field in fields:
+        if entity.get(field) is None:
+            name = "Untitled" if entity.get("name") is None else entity.get("name")
+            print("Error:", typeof, name,
+                  "is missing mandatory field", field)
+            return False
+        events = entity.get("events")
+        if events is not None:
+            for event in events:
+                if not verify(event, event_non_nullable, "event"):
+                    return False
+        tags = entity.get("tags")
+        if tags is not None:
+            for tag in tags:
+                if tag not in tags_enum:
+                    print("Error:", typeof, entity.get(
+                        "name"), "has illegal tag", tag)
+                    return False
+    return True
+
+
+def insert_clubs():
+    club_list = read_clubs(filepath)
+    if club_list is None:
+        print("Something failed; aborting...")
+        return
+    conn = connect()
+    if conn is None:
+        return
+    insert_tags()
+    try:
+        while True:
+            club = next(club_list)
+            club["id"] = str(uuid.uuid4())
+            # pp.pprint(club)
+            with conn.cursor() as cursor:
+                cursor.execute("INSERT INTO club VALUES (%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s);",
+                               (club.get("id"), datetime.now(), datetime.now(), club.get("name"),
+                                club.get("description"), club.get("size"), club.get(
+                                    "banner_image"), club.get("icon_image"),
+                                club.get("facebook"), club.get("twitter"), club.get("instagram"), club.get("website")))
+                events = club.get("events")
+                if (events is not None):
+                    for event in events:
+                        event["id"] = str(uuid.uuid4())
+                        cursor.execute(
+                            "INSERT INTO event VALUES (%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s,%s);",
+                            (event.get("id"), datetime.now(), datetime.now(), event.get("name"), event.get("description"),
+                             event.get("url"), event.get("banner_image"),
+                             event.get("start_time"), event.get("end_time"),
+                             event.get("facebook"), event.get("twitter"), event.get("instagram")))
+                        cursor.execute(
+                            "INSERT INTO club_events VALUES (%s, %s);", (club.get("id"), event.get("id")))
+                        tags = event.get("tags")
+                        if tags is not None:
+                            for tag in tags:
+                                cursor.execute(
+                                    "INSERT INTO tag_events VALUES (%s, %s);", (tag_entities.get(tag), event.get("id")))
```

Comment:
This is a "request in a loop", which could cause some serious performance (and thus finance) issues. The resolution is to do as much as possible in a single `execute` statement. There seems to be a variety of ways to do this, according to [this SO thread](https://stackoverflow.com/questions/8134602/psycopg2-insert-multiple-rows-with-one-query)